File: console/src/features/tokens/dto/token-response.ts
export interface TokenResponse {
  pk: string;
  name: string;
  symbol: string;
  decimals: number;
  total_supply: number;
  circulating_supply: number;
  description?: string;
  created_at: number;
  updated_at: number;
}

export interface ListTokensResponse {
  tokens: TokenResponse[];
}

File: app/src/features/tokens/dto/token_response.rs
use serde::{Deserialize, Serialize};
use crate::common::types::Partition;

#[derive(Debug, Serialize, Deserialize)]
pub struct TokenResponse {
    pub pk: Partition,
    pub name: String,
    pub symbol: String,
    pub decimals: u8,
    pub total_supply: i64,
    pub circulating_supply: i64,
    pub description: Option<String>,
    pub created_at: i64,
    pub updated_at: i64,
}
